packages/rspack/src/builtin-plugin/lazy-compilation/middleware.ts

```diff
@@ -37,19 +37,104 @@ const getFullServerUrl = ({ serverUrl, prefix }: LazyCompilationOptions) => {
 };
 
 export const lazyCompilationMiddleware = (
-	compiler: Compiler,
-	userOptions: LazyCompilationOptions | boolean = {}
+	compiler: Compiler | MultiCompiler,
+	rawOptions?: LazyCompilationOptions | boolean
 ): Middleware => {
+	let userOptions = rawOptions;
 	if (userOptions === false) {
 		return noop;
 	}
 
-	const options = userOptions === true ? {} : userOptions;
+	if (userOptions === true) {
+		userOptions = {};
+	}
+
+	if (compiler instanceof MultiCompiler) {
+		const middlewareByCompiler: Map<string, Middleware> = new Map();
+
+		let i = 0;
+		for (const c of compiler.compilers) {
+			if (!c.options.experiments.lazyCompilation) {
```

Comment:
Where is `c.options.experiments.lazyCompilation` been set

Reply:
We should force users to set `experiments.lazyCompilation` instead of passing `rawOptions` to the middlewware